### PR TITLE
Use xpath query handler

### LINKED ISSUE #2085: Add missing Xpath Selector engine

### PR DESCRIPTION
**Pull request by kblok**
closes #2085

### CODE REVIEW

File: lib/PuppeteerSharp/CustomQueriesManager.cs [hunk 1]
```diff
@@ -155,21 +190,23 @@ internal void RegisterCustomQueryHandler(string name, CustomQueryHandler queryHa
...
-            {
-                return (selector, _defaultHandler);
-            }
-
-            var name = customQueryHandlerMatch.Groups["query"].Value;
-            var updatedSelector = customQueryHandlerMatch.Groups["selector"].Value;
+            var handlers = _internalQueryHandlers.Concat(_queryHandlers);
 
-            if (!_queryHandlers.TryGetValue(name, out var queryHandler))
+            foreach (var kv in handlers)
```
Comment (jnyrup) [inline lib/PuppeteerSharp/CustomQueriesManager.cs:195]: Use 4 spaces instead of 2 🧹

File: lib/PuppeteerSharp/CustomQueriesManager.cs [hunk 2]
```diff
@@ -10,6 +10,8 @@ namespace PuppeteerSharp
 {
     internal class CustomQueriesManager
     {
+        private static readonly string[] CustomQuerySeparators = new[] { "=", "/" };
+        private readonly Dictionary<string, PuppeteerQueryHandler> _internalQueryHandlers;
```
Comment (jnyrup) [inline lib/PuppeteerSharp/CustomQueriesManager.cs:14]: Suggestion:
Besides `_defaultHandler`, we don't need to store the internal handlers in separate fields as they are only used in the constructor to initialize `_internalQueryHandlers`.

By changing them from fields to private static creator methods, we can move initialization of `_internalQueryHandlers` to declaration and remove the then empty public constructor.

```cs
private readonly Dictionary<string, PuppeteerQueryHandler> _internalQueryHandlers = new()
{
    ["aria"] = AriaHandler(),
    ["pierce"] = PierceHandler(),
    ["text"] = TextQueryHandler(),
    ["xpath"] = XpathHandler(),
};
```

File: lib/PuppeteerSharp/CustomQueriesManager.cs (hunk 1 above)
Comment (kblok) [reply lib/PuppeteerSharp/CustomQueriesManager.cs:195]: ohhh just merged! Let's create a follow up PR

Comment (jnyrup) [reply lib/PuppeteerSharp/CustomQueriesManager.cs:195]: I'll create a PR right away

Comment (kblok) [reply lib/PuppeteerSharp/CustomQueriesManager.cs:195]: I have one ready to go :)

**Pull request merged**